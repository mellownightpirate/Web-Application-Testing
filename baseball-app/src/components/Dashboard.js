import React from 'react'

export default function Dashboard(props) {

return (
<div className="container">
<button data-testid="baller" className="button" onClick={props.ball}>Ball</button>
   <button data-testid="striker" className="button" onClick={props.strike}>Strike</button>
   <button data-testid="hitter" className="button" onClick={props.hit}>Hit</button>
   <button data-testid="fouler" className="button" onClick={props.foul}>Foul</button>
   <button data-testid="inns" className="button" onClick={props.inning}>Inning </button>
   <button data-testid="outs" className="button" onClick={props.out}>Out </button>
</div>
)
}