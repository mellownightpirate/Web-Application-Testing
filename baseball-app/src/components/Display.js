import React from 'react';

export default function Display(props) {
    
    return (
        <div className="container">
            <h2 data-testid="displayBalls">Balls : {props.balls}</h2>
            <h2 data-testid="displayStrikes">Strikes : {props.strikes}</h2>
            <h2 data-testid="displayOuts">Outs : {props.outs}</h2>
            <h2 data-testid="displayInnings">Innings : {props.innings}</h2>
        </div>
    )
} 