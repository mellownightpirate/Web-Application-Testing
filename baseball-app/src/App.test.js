import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { fireEvent, cleanup } from '@testing-library/react';
import App from './App';
import Dashboard, {strikes} from './components/Dashboard';
import Display, {displayStrikes} from './components/Display';

afterEach(cleanup);
it('match the snapshot', () =>{
  const { asFragment } = rtl.render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

test('renders display', () => {
  const wrapper = rtl.render(<Display/>)
  const element = wrapper.getByText(/strike/i)
  expect(element).toBeInTheDocument();
});

test('renders dashboard', () => {
  const wrapper = rtl.render(<Dashboard/>)
  const element = wrapper.getByText(/hit/i)
  expect(element).toBeInTheDocument();
});


test('strikes are less than 3', ()=>{
  const { getByTestId } = rtl.render(<App/>);
  const displayStrikes = getByTestId('displayStrikes');
  const striker = getByTestId('striker');
  expect(displayStrikes.textContent).toBe('Strikes : 0')
   fireEvent.click(striker)
   expect(displayStrikes.textContent).toBe('Strikes : 1')
   fireEvent.click(striker)
   expect(displayStrikes.textContent).toBe('Strikes : 2')
   fireEvent.click(striker)
   expect(displayStrikes.textContent).toBe('Strikes : 0')
});

test('balls are less than 3', ()=>{
  const { getByTestId } = rtl.render(<App/>);
  const displayBalls = getByTestId('displayBalls');
  const baller = getByTestId('baller');
  expect(displayBalls.textContent).toBe('Balls : 0')
   fireEvent.click(baller)
   expect(displayBalls.textContent).toBe('Balls : 1')
   fireEvent.click(baller)
   expect(displayBalls.textContent).toBe('Balls : 2')
   fireEvent.click(baller)
   expect(displayBalls.textContent).toBe('Balls : 3')
   fireEvent.click(baller)
   expect(displayBalls.textContent).toBe('Balls : 0')
   fireEvent.click(baller)
});