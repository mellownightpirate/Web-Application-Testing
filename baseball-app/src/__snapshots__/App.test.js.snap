// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1>
      Player's At Bat
    </h1>
    <div
      class="container"
    >
      <h2
        data-testid="displayBalls"
      >
        Balls : 0
      </h2>
      <h2
        data-testid="displayStrikes"
      >
        Strikes : 0
      </h2>
      <h2
        data-testid="displayOuts"
      >
        Outs : 0
      </h2>
      <h2
        data-testid="displayInnings"
      >
        Innings : 0
      </h2>
    </div>
    <div
      class="container"
    >
      <button
        class="button"
        data-testid="baller"
      >
        Ball
      </button>
      <button
        class="button"
        data-testid="striker"
      >
        Strike
      </button>
      <button
        class="button"
        data-testid="hitter"
      >
        Hit
      </button>
      <button
        class="button"
        data-testid="fouler"
      >
        Foul
      </button>
      <button
        class="button"
        data-testid="inns"
      >
        Inning 
      </button>
      <button
        class="button"
        data-testid="outs"
      >
        Out 
      </button>
    </div>
  </div>
</DocumentFragment>
`;
